lib/pages/surveydetail/survey_detail_view_model.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_survey/models/survey_detail.dart';
+import 'package:flutter_survey/pages/surveydetail/survey_detail_state.dart';
+import 'package:flutter_survey/pages/uimodel/survey_ui_model.dart';
+import 'package:flutter_survey/usecase/base/base_use_case.dart';
+import 'package:flutter_survey/usecase/get_survey_detail_use_case.dart';
+import 'package:rxdart/subjects.dart';
+
+class SurveyDetailViewModel extends StateNotifier<SurveyDetailState> {
+  final GetSurveyDetailUseCase _getSurveyDetailUseCase;
+
+  SurveyDetailViewModel(this._getSurveyDetailUseCase)
+      : super(const SurveyDetailState.init());
+
+  final BehaviorSubject<SurveyUiModel> _surveySubject = BehaviorSubject();
+
+  Stream<SurveyUiModel> get surveyStream => _surveySubject.stream;
+
+  Future<void> loadSurveyDetail(SurveyUiModel survey) async {
+    // Load initial survey
+    state = const SurveyDetailState.loading();
+    _surveySubject.add(survey);
+    state = const SurveyDetailState.success();
+
+    // Fetch latest survey detail
+    state = const SurveyDetailState.loading();
```

Comment:
Do we really need these updates, it mainly changes instantly right? 🤔

Reply:
@markgravity we need this to display our survey data from Home screen list. We don't need to wait until having a latest one in detail for some basic info 🤓